src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsPostEquilibrationStep.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SolidMechanicsPostEquilibrationStep.hpp
+ */
+
+#ifndef SRC_CORECOMPONENTS_PHYSICSSOLVERS_SOLIDMECHANICS_SOLIDMECHANICSPOSTEQUILIBRATIONSTEP_HPP_
+#define SRC_CORECOMPONENTS_PHYSICSSOLVERS_SOLIDMECHANICS_SOLIDMECHANICSPOSTEQUILIBRATIONSTEP_HPP_
+
+#include "events/tasks/TaskBase.hpp"
+
+namespace geosx
+{
+
+class SolidMechanicsLagrangianFEM;
+
+/**
+ * @class SolidMechanicsPostEquilibrationStep
+ *
+ * This class assumes that a equilibration step has been performed at the start of the simulation.
+ * The role of the class is to implement a task that resets the total displacements to zero after the equilibration step.
```

Comment:
Note that it does also reset velocity and incremental displacement.
If I were you, I would not document this level of detail, but the position of `SolidMechanicsLagrangianFEM` in t]he global process instead.

More important, if this is part of a global algorithm for plasticity, why would you separate this task from the solvers themselves?

Reply:
Done, it is now better documented.

The reason why this is separated from the solver and placed within a `Task` is that this makes it callable from a `SoloEvent`. The `Task` itself really does not do much in fact (no kernels), it is just resetting 3 fields and turning on/off a flag. So I think the end result is quite clean.